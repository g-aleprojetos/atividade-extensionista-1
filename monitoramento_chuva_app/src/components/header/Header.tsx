import React, {useState} from 'react';
import {Nav} from 'components/nav';
import * as S from './Header.styles';
import {Popup} from 'components/popup/popupConfirma';
import {ILoginResponse} from 'resources/interfaces';
import {useAutenticadoContext} from 'context/autenticador';
import {useNavigate} from 'react-router-dom';
import rotas from 'resources/rotas';

export const Header = () => {
  const navigate = useNavigate();
  const {handleAutenticado} = useAutenticadoContext();
  const [openLogout, setOpenLogout] = useState<boolean>(false);
  const handleLogout = () => {
    setOpenLogout(prev => !prev);
  };

  const handleOnConfirmarLogout = () => {
    handleLogout;
    navigate(rotas.Login);
    localStorage.clear();
    const logOut: ILoginResponse = {
      auth: false,
      accessToken: '',
      refreshToken: {id: '', expiresIn: 0, usuarioId: ''},
    };
    handleAutenticado(logOut);
  };

  return (
    <>
      <S.ContainerHeader data-testid={'test_header'}>
        <S.ContainerContent>
          <Nav handleLogout={handleLogout} />
        </S.ContainerContent>
      </S.ContainerHeader>
      {openLogout && (
        <Popup
          textTitle="Deseja fazer Logout?"
          onConfirma={handleOnConfirmarLogout}
          onClickClose={handleLogout}
        />
      )}
    </>
  );
};
