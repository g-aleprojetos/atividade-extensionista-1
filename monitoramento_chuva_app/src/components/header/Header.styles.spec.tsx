import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import * as S from './Header.styles';

describe('Header.styles', () => {
  test('ContainerHeader DEVE ser igual ao snapshot', () => {
    const containerHeader = renderer.create(<S.ContainerHeader />).toJSON();
    expect(containerHeader).toMatchSnapshot();
  });

  test('ContainerContent DEVE ser igual ao snapshot', () => {
    const containerContent = renderer.create(<S.ContainerContent />).toJSON();
    expect(containerContent).toMatchSnapshot();
  });

  test('ContainerContent DEVE ser igual ao snapshot', () => {
    const containerLogout = renderer.create(<S.ContainerLogout />).toJSON();
    expect(containerLogout).toMatchSnapshot();
  });
});
