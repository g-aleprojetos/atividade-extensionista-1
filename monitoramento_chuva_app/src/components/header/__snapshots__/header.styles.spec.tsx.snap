// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.styles ContainerContent DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  width: 100%;
  height: 2.125rem;
  top: 3.438rem;
}

<div
  className="c0"
/>
`;

exports[`Header.styles ContainerContent DEVE ser igual ao snapshot 2`] = `
.c0 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  background-color: #00000080;
}

<div
  className="c0"
/>
`;

exports[`Header.styles ContainerHeader DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 9rem;
  background-color: #3C3C3C;
}

<div
  className="c0"
/>
`;
